frontend/src/framework/internal/components/SelectEnsemblesDialog/selectEnsemblesDialog.tsx

```diff
@@ -198,14 +249,33 @@ export const SelectEnsemblesDialog: React.FC<SelectEnsemblesDialogProps> = (prop
                                 errorComponent={<div className="text-red-500">Error loading cases</div>}
                                 loadingComponent={<CircularProgress />}
                             >
-                                <Select
+                                <div className="flex justify-end gap-4 items-center">
+                                    <span className="flex-grow text-sm text-slate-500">
+                                        Select from {caseOpts.length} cases
+                                    </span>
+                                    <Label position="right" text="Keep" title="Show only cases marked as keep">
+                                        <Switch
+                                            checked={casesFilteringOptions.keep}
+                                            onChange={handleKeepCasesSwitchChange}
+                                        />
+                                    </Label>
+                                    <Label position="right" text="My cases" title="Show only cases that I created">
```

Comment:
I'm fine with would current label title. If we want to align with SUMO (and shorten the text) it would be something like "Show only my cases"
![image](https://github.com/equinor/webviz/assets/31612826/b548d508-9dbf-4982-9a2e-f3f3c84136cc)